### ISSUE TITLE
Non-standard callback arguments

### ISSUE THREAD

**Issue by raine** [sonicsnes/node-gamedig#51]
Why doesn't this library return error as the first argument and result as second argument, as is the standard in node?

I'm asking because I tried promisifying the library and got an error because promisification libraries expect that first argument is an error, as they should.

**Comment by sonicsnes** [comment]
Not for any good reason unfortunately. This code was originally written to be plugged into an older application which expected return values of this type (coincidentally that application has also converted to promises and now uses gamedig through a legacy wrapper).

We may be able to support both promises and nodejs-standard callbacks by sniffing the argument parameter counts. I'll give it a shot tonight if I have a spare moment.

**Comment by raine** [comment]
If that is not possible, a new major version with correct style could be released and nothing should break.

**Comment by sonicsnes** [comment]
Fixed in 7b9fe2161cc54c7d7c36a50105aec5b8c6f12862 v0.2.26

Give it a shot!

**Comment by raine** [comment]
Cheers, it works.

```sh
cat servers.txt | \
  R -r --js --import gamedig \
  'split(":")' \
  '([host, port]) => gamedig.query({ type: "rust", host: host, port: parseInt(port) })' \
  '(p) => p.catch(() => {})' | \
  R --slurp -o table --compact 'map(pick(["name", "map"]))'
┌───────────────────────────────────────────────────┬────────────────┐
│ name                                              │ map            │
├───────────────────────────────────────────────────┼────────────────┤
│ Armoton Suomi - Merciless Finland - armoton.fi    │ Procedural Map │
│ Pelipalvelin.com Rust server                      │ Procedural Map │
│ Hubu ServeBeer                                    │ Procedural Map │
│ [FI/EU] Rakeservers.net - Vanilla - Last wipe 4.3 │ Procedural Map │
│ SE/DK/NO/FI - Road to Valhalla                    │ Procedural Map │
│ Firestarters [FI/EU]                              │ Procedural Map │
│ FRT Rust Server                                   │ Procedural Map │
│ FIN Rust public server                            │ Procedural Map │
│ Entajuu.com Rust01                                │ Procedural Map │
└───────────────────────────────────────────────────┴────────────────┘
```
